papis/commands/add.py

```diff
@@ -247,25 +247,28 @@ def run(paths: List[str],
     if citations is None:
         citations = []
 
+    for p in paths:
+        if not os.path.exists(p):
+            raise FileNotFoundError(f"File '{p}' not found")
+
     import tempfile
 
     in_documents_paths = paths
     temp_dir = tempfile.mkdtemp()
-
-    for p in in_documents_paths:
-        if not os.path.exists(p):
-            raise FileNotFoundError(f"File '{p}' not found")
+    tmp_document = papis.document.Document(folder=temp_dir, data=data)
 
```

Comment:
I would add the `papis_id` computation right here, i.e. 
```suggestion
    tmp_document = papis.document.Document(folder=temp_dir, data=data)
	papis.database.get().maybe_compute_id(tmp_document)
```
Then it can be used by the ref and probably don't need to move any of the other things around either? Is there any issue with doing that? Maybe complain if `data` is empty?

Reply:
done